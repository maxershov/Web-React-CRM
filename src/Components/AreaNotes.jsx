import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { ChangeProfileValue, addNewDayDataToJSON } from '../App';


const AreaNotes = (props) => {
  const { notesValue, type, dayObject } = props;
  const [notesData, setNotesData] = useState(notesValue);
  const { codeLink } = useParams();

  const saveNotes = (event) => {
    if (event.key === 'Enter' || event.target.id === 'clickNotes') {
      if (type === 'PERSON') {
        ChangeProfileValue(codeLink, notesData, "notes")
      } else if (type === 'DAY_DATA') {
        dayObject.notes = notesData;
        addNewDayDataToJSON(dayObject);
      }
    }
  }
  useEffect(() => {
    setNotesData(notesValue);
  }, [notesValue]);
  return (
    <>
      <textarea onChange={event => setNotesData(event.target.value)} onKeyDown={saveNotes} value={notesData} />
      <button type="button" id="clickNotes" onClick={saveNotes}>Изменить</button>
    </>
  )
}

export default AreaNotes;