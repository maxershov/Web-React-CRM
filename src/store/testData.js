import { format } from 'date-fns'
export const dayData = [
  {
    date: format(new Date(),"dd-MM-yyyy"),
    notes: "Заметки меняются при изменении даты",
    history: []
  }
];

export const activityData = [
  {
    code: "1",
    activity: [
      {
        date: "21-12-2019",
        time: "12:25:40",
        type: "Продажа",
        person: "",
        amount: "Продажа доп.услуги"
      }
    ]
  },
  {
    code: "2",
    activity: [
      {
        date: "21-12-2019",
        time: "12:25:40",
        type: "Продажа",
        person: "",
        amount: "Продажа доп.услуги"
      }
    ]
  },
  {
    code: "3",
    activity: [
      {
        date: "21-12-2019",
        time: "12:25:40",
        type: "Продажа",
        person: "",
        amount: "Продажа доп.услуги"
      }
    ]
  },
  {
    code: "4",
    activity: [
      {
        date: "21-12-2019",
        time: "12:25:40",
        type: "Продажа",
        person: "",
        amount: "Продажа доп.услуги"
      }
    ]
  },
  {
    code: "5",
    activity: []
  },
  {
    code: "6",
    activity: []
  },
  {
    code: "7",
    activity: []
  },
  {
    code: "8",
    activity: []
  },
  {
    code: "9",
    activity: []
  },
  {
    code: "10",
    activity: []
  },
  {
    code: "11",
    activity: []
  },
  {
    code: "12",
    activity: []
  },
  {
    code: "13",
    activity: []
  },
  {
    code: "14",
    activity: []
  },
  {
    code: "15",
    activity: []
  },
  {
    code: "16",
    activity: []
  },
  {
    code: "17",
    activity: []
  },
  {
    code: "18",
    activity: []
  },
  {
    code: "19",
    activity: []
  },
  {
    code: "20",
    activity: []
  },
  {
    code: "21",
    activity: []
  },
  {
    code: "22",
    activity: []
  },
  {
    code: "23",
    activity: []
  },
  {
    code: "24",
    activity: []
  },
  {
    code: "25",
    activity: []
  },
  {
    code: "26",
    activity: []
  },
  {
    code: "27",
    activity: []
  },
  {
    code: "28",
    activity: []
  },
  {
    code: "29",
    activity: []
  },
  {
    code: "30",
    activity: []
  },
  {
    code: "31",
    activity: []
  },
  {
    code: "32",
    activity: []
  },
  {
    code: "33",
    activity: []
  },
  {
    code: "34",
    activity: []
  },
  {
    code: "35",
    activity: []
  },
  {
    code: "36",
    activity: []
  },
  {
    code: "37",
    activity: []
  },
  {
    code: "38",
    activity: []
  },
  {
    code: "39",
    activity: []
  },
  {
    code: "40",
    activity: []
  },
  {
    code: "41",
    activity: []
  },
  {
    code: "42",
    activity: []
  },
  {
    code: "43",
    activity: []
  },
  {
    code: "44",
    activity: []
  },
  {
    code: "45",
    activity: []
  },
  {
    code: "46",
    activity: []
  },
  {
    code: "47",
    activity: []
  },
  {
    code: "48",
    activity: []
  },
  {
    code: "49",
    activity: []
  },
  {
    code: "50",
    activity: []
  }
];

export const personData = [
  {
    personName: "Murphy Mccarthy",
    contract: "МИНИГРУППА",
    dateBirth: "17-10-1979",
    telNum: "1-256-471-0997",
    code: "1",
    autoMonth: 4,
    remain: "10",
    days: "",
    photoId: 3,
    rent: "21-04-2020",
    deposite: 3699,
    notes: "diam. Pellentesque habitant morbi tristique"
  },
  {
    personName: "Garrett Howe",
    contract: "6 ГРУПП",
    dateBirth: "09-06-1994",
    telNum: "1-843-886-7481",
    code: "2",
    autoMonth: 7,
    remain: "",
    days: "21-05-2020",
    photoId: 6,
    rent: "26-10-2019",
    deposite: 1085,
    notes: "magna. Phasellus dolor elit, pellentesque"
  },
  {
    personName: "Hilel Castaneda",
    contract: "3 ГРУПП",
    dateBirth: "11-03-1988",
    telNum: "1-737-288-1405",
    code: "3",
    autoMonth: 2,
    remain: "",
    days: "13-02-2020",
    photoId: 0,
    rent: "26-09-2019",
    deposite: 3669,
    notes: "ut quam vel sapien imperdiet"
  },
  {
    personName: "Rhoda Webb",
    contract: "1 СВОБ",
    dateBirth: "02-04-1967",
    telNum: "1-870-541-1034",
    code: "4",
    autoMonth: 8,
    remain: "",
    days: "27-06-2020",
    photoId: 0,
    rent: "15-10-2019",
    deposite: 303,
    notes: "auctor ullamcorper, nisl arcu iaculis"
  },
  {
    personName: "Kylie Blackburn",
    contract: "ЛИД",
    dateBirth: "04-07-1981",
    telNum: "1-562-298-7788",
    code: "5",
    autoMonth: 9,
    remain: 12,
    days: "24-10-2019",
    photoId: 8,
    rent: "12-01-2020",
    deposite: 3724,
    notes: "Curabitur sed tortor. Integer aliquam"
  },
  {
    personName: "Autumn Swanson",
    contract: "ЛИД",
    dateBirth: "03-10-1983",
    telNum: "1-880-353-9575",
    code: "6",
    autoMonth: 1,
    remain: 9,
    days: "17-11-2019",
    photoId: 0,
    rent: "02-11-2019",
    deposite: 4289,
    notes: "In scelerisque scelerisque dui. Suspendisse"
  },
  {
    personName: "Montana Kaufman",
    contract: "СОТРУДНИК",
    dateBirth: "21-11-1981",
    telNum: "1-973-533-7140",
    code: "7",
    autoMonth: 8,
    remain: 10,
    days: "25-10-2020",
    photoId: 7,
    rent: "08-10-2019",
    deposite: 1588,
    notes: "Praesent interdum ligula eu enim."
  },
  {
    personName: "Otto Ryan",
    contract: "НЕТ",
    dateBirth: "27-07-1963",
    telNum: "1-892-415-9444",
    code: "8",
    autoMonth: 3,
    remain: 8,
    days: "29-05-2019",
    photoId: 10,
    rent: "03-09-2019",
    deposite: 4755,
    notes: "aliquet, sem ut cursus luctus,"
  },
  {
    personName: "Karleigh Hurley",
    contract: "НЕТ",
    dateBirth: "23-01-1954",
    telNum: "1-844-835-1769",
    code: "9",
    autoMonth: 1,
    remain: 13,
    days: "08-03-2019",
    photoId: 0,
    rent: "25-09-2019",
    deposite: 1454,
    notes: "placerat, orci lacus vestibulum lorem,"
  },
  {
    personName: "Rebekah English",
    contract: "СОТРУДНИК",
    dateBirth: "15-05-1978",
    telNum: "1-893-501-2845",
    code: "10",
    autoMonth: 8,
    remain: 22,
    days: "08-01-2021",
    photoId: 9,
    rent: "01-12-2019",
    deposite: 2972,
    notes: "luctus et ultrices posuere cubilia"
  },
  {
    personName: "Francis Cantu",
    contract: "ЛИД",
    dateBirth: "29-02-1972",
    telNum: "1-218-584-2826",
    code: "11",
    autoMonth: 7,
    remain: 24,
    days: "21-12-2019",
    photoId: 0,
    rent: "21-02-2019",
    deposite: 2790,
    notes: "velit. Quisque varius. Nam porttitor"
  },
  {
    personName: "Raja Rowland",
    contract: "12 СВОБ",
    dateBirth: "05-11-1991",
    telNum: "1-630-349-1486",
    code: "12",
    autoMonth: 8,
    remain: "",
    days: "07-05-2019",
    photoId: 0,
    rent: "16-03-2019",
    deposite: 3750,
    notes: "ac facilisis facilisis, magna tellus"
  },
  {
    personName: "Mufutau Price",
    contract: "3 КФ",
    dateBirth: "26-10-1961",
    telNum: "1-571-985-3972",
    code: "13",
    autoMonth: 5,
    remain: "",
    days: "26-04-2019",
    photoId: 0,
    rent: "07-09-2020",
    deposite: -123,
    notes: "sed, facilisis vitae, orci. Phasellus"
  },
  {
    personName: "Tate Kerr",
    contract: "НЕТ",
    dateBirth: "25-07-1983",
    telNum: "1-736-691-3234",
    code: "14",
    autoMonth: 1,
    remain: 9,
    days: "25-01-2020",
    photoId: 0,
    rent: "14-10-2019",
    deposite: 1388,
    notes: "non, vestibulum nec, euismod in,"
  },
  {
    personName: "Nichole Tran",
    contract: "СОТРУДНИК",
    dateBirth: "09-11-1980",
    telNum: "1-977-432-7973",
    code: "15",
    autoMonth: 5,
    remain: 23,
    days: "11-10-2020",
    photoId: 11,
    rent: "22-01-2021",
    deposite: 4916,
    notes: "arcu. Vestibulum ut eros non"
  },
  {
    personName: "Kibo Hayden",
    contract: "НЕТ",
    dateBirth: "11-07-1987",
    telNum: "1-701-574-1974",
    code: "16",
    autoMonth: 7,
    remain: 7,
    days: "09-01-2020",
    photoId: 0,
    rent: "30-11-2019",
    deposite: 3092,
    notes: "odio a purus. Duis elementum,"
  },
  {
    personName: "McKenzie Phillips",
    contract: "1 СВОБ",
    dateBirth: "22-09-1956",
    telNum: "1-506-702-6936",
    code: "17",
    autoMonth: 8,
    remain: "",
    days: "05-08-2019",
    photoId: 4,
    rent: "06-04-2019",
    deposite: 351,
    notes: "elementum purus, accumsan interdum libero"
  },
  {
    personName: "TaShya Moore",
    contract: "1 СВОБ",
    dateBirth: "10-10-1965",
    telNum: "1-115-227-9451",
    code: "18",
    autoMonth: 8,
    remain: "",
    days: "04-11-2019",
    photoId: 0,
    rent: "07-11-2020",
    deposite: 1003,
    notes: "fringilla cursus purus. Nullam scelerisque"
  },
  {
    personName: "Holly Bradley",
    contract: "6 КФ",
    dateBirth: "",
    telNum: "",
    code: "19",
    autoMonth: 8,
    remain: "",
    days: "17-02-2019",
    photoId: 2,
    rent: "26-02-2019",
    deposite: 2104,
    notes: "lectus ante dictum mi, ac"
  },
  {
    personName: "Xander Moses",
    contract: "МИНИГРУППА",
    dateBirth: "",
    telNum: "1-942-292-3642",
    code: "20",
    autoMonth: 9,
    remain: 14,
    days: "30-11-2019",
    photoId: 5,
    rent: "07-01-2020",
    deposite: 3493,
    notes: "sapien molestie orci tincidunt adipiscing."
  },
  {
    personName: "Amaya Valentine",
    contract: "ПТ",
    dateBirth: "10-11-1953",
    telNum: "1-740-709-8222",
    code: "21",
    autoMonth: 5,
    remain: 18,
    days: "26-11-2019",
    photoId: 0,
    rent: "06-06-2020",
    deposite: 3790,
    notes: "porttitor eros nec tellus. Nunc"
  },
  {
    personName: "Thane Alvarado",
    contract: "ПТ",
    dateBirth: "01-10-1982",
    telNum: "1-872-104-7996",
    code: "22",
    autoMonth: 6,
    remain: 22,
    days: "03-02-2019",
    photoId: 0,
    rent: "25-03-2020",
    deposite: 1335,
    notes: "lobortis augue scelerisque mollis. Phasellus"
  },
  {
    personName: "Coby Mejia",
    contract: "ПТ",
    dateBirth: "13-11-1976",
    telNum: "1-372-186-7017",
    code: "23",
    autoMonth: 8,
    remain: 19,
    days: "06-03-2020",
    photoId: 0,
    rent: "05-02-2019",
    deposite: 2935,
    notes: "non enim commodo hendrerit. Donec"
  },
  {
    personName: "Travis Mccullough",
    contract: "ПТ",
    dateBirth: "15-09-1987",
    telNum: "1-743-426-2899",
    code: "24",
    autoMonth: 8,
    remain: 4,
    days: "19-08-2020",
    photoId: 0,
    rent: "04-05-2020",
    deposite: 3009,
    notes: "conubia nostra, per inceptos hymenaeos."
  },
  {
    personName: "George Kaufman",
    contract: "3 КФ",
    dateBirth: "03-02-1948",
    telNum: "1-140-514-2906",
    code: "25",
    autoMonth: 8,
    remain: "",
    days: "27-10-2020",
    photoId: 0,
    rent: "13-07-2019",
    deposite: 1141,
    notes: "nascetur ridiculus mus. Aenean eget"
  },
  {
    personName: "Adrienne Craft",
    contract: "ЛИД",
    dateBirth: "02-05-1981",
    telNum: "1-974-412-0982",
    code: "26",
    autoMonth: 5,
    remain: 3,
    days: "29-11-2019",
    photoId: 0,
    rent: "24-10-2019",
    deposite: 4554,
    notes: "tellus. Suspendisse sed dolor. Fusce"
  },
  {
    personName: "Cain Roberson",
    contract: "НЕТ",
    dateBirth: "10-04-1965",
    telNum: "1-747-108-3057",
    code: "27",
    autoMonth: 1,
    remain: 25,
    days: "14-12-2019",
    photoId: 0,
    rent: "14-05-2019",
    deposite: 1530,
    notes: "Vestibulum ante ipsum primis in"
  },
  {
    personName: "Ivana Carey",
    contract: "СОТРУДНИК",
    dateBirth: "27-12-1944",
    telNum: "1-822-483-5692",
    code: "28",
    autoMonth: 9,
    remain: 8,
    days: "15-04-2019",
    photoId: 0,
    rent: "23-02-2020",
    deposite: 2230,
    notes: "taciti sociosqu ad litora torquent"
  },
  {
    personName: "Merritt Becker",
    contract: "12 СВОБ",
    dateBirth: "07-10-1953",
    telNum: "1-414-164-3453",
    code: "29",
    autoMonth: 1,
    remain: "",
    days: "28-01-2019",
    photoId: 0,
    rent: "01-02-2019",
    deposite: -205,
    notes: "velit. Cras lorem lorem, luctus"
  },
  {
    personName: "Shay Randolph",
    contract: "ПТ",
    dateBirth: "28-03-2004",
    telNum: "1-498-448-2146",
    code: "30",
    autoMonth: 3,
    remain: 10,
    days: "21-02-2020",
    photoId: 0,
    rent: "26-04-2019",
    deposite: 82,
    notes: "rutrum lorem ac risus. Morbi"
  },
  {
    personName: "Isaiah Sweet",
    contract: "3 КФ",
    dateBirth: "27-01-1981",
    telNum: "1-404-934-8475",
    code: "31",
    autoMonth: 9,
    remain: "",
    days: "01-12-2019",
    photoId: 0,
    rent: "22-01-2020",
    deposite: 2543,
    notes: "cursus luctus, ipsum leo elementum"
  },
  {
    personName: "Caleb Washington",
    contract: "12 КФ",
    dateBirth: "02-03-1975",
    telNum: "1-437-259-5837",
    code: "32",
    autoMonth: 5,
    remain: 5,
    days: "31-10-2020",
    photoId: 0,
    rent: "03-02-2020",
    deposite: 3991,
    notes: "quis diam. Pellentesque habitant morbi"
  },
  {
    personName: "Christen Dominguez",
    contract: "МИНИГРУППА",
    dateBirth: "30-03-1974",
    telNum: "1-266-983-7881",
    code: "33",
    autoMonth: 5,
    remain: 19,
    days: "02-01-2021",
    photoId: 0,
    rent: "17-12-2019",
    deposite: 564,
    notes: "Fusce mollis. Duis sit amet"
  },
  {
    personName: "Reese Barry",
    contract: "3 ТА",
    dateBirth: "12-01-1964",
    telNum: "1-169-703-3860",
    code: "34",
    autoMonth: 9,
    remain: "",
    days: "05-10-2020",
    photoId: 0,
    rent: "17-08-2019",
    deposite: 2498,
    notes: "posuere at, velit. Cras lorem"
  },
  {
    personName: "Flynn Moran",
    contract: "3 ТА",
    dateBirth: "23-01-1982",
    telNum: "1-623-206-4393",
    code: "35",
    autoMonth: 7,
    remain: "",
    days: "29-02-2020",
    photoId: 0,
    rent: "05-09-2019",
    deposite: 2654,
    notes: "et ipsum cursus vestibulum. Mauris"
  },
  {
    personName: "Tashya Faulkner",
    contract: "1 СВОБ",
    dateBirth: "06-06-1948",
    telNum: "1-726-285-5889",
    code: "36",
    autoMonth: 6,
    remain: "",
    days: "27-09-2020",
    photoId: 0,
    rent: "26-07-2019",
    deposite: 2232,
    notes: "elit, a feugiat tellus lorem"
  },
  {
    personName: "April Lynn",
    contract: "ПОСЕЩЕНИЯ",
    dateBirth: "23-08-1953",
    telNum: "1-646-749-0720",
    code: "37",
    autoMonth: 6,
    remain: 22,
    days: "05-03-2019",
    photoId: 0,
    rent: "31-12-2020",
    deposite: 2731,
    notes: "tincidunt pede ac urna. Ut"
  },
  {
    personName: "Pearl Nguyen",
    contract: "3 КФ",
    dateBirth: "25-12-1978",
    telNum: "1-233-110-5293",
    code: "38",
    autoMonth: 7,
    remain: "",
    days: "14-08-2020",
    photoId: 0,
    rent: "14-08-2019",
    deposite: 2941,
    notes: "eget metus eu erat semper"
  },
  {
    personName: "Lael Cervantes",
    contract: "ЛИД",
    dateBirth: "26-07-1995",
    telNum: "1-825-543-2207",
    code: "39",
    autoMonth: 4,
    remain: 7,
    days: "22-03-2019",
    photoId: 0,
    rent: "23-12-2019",
    deposite: 1166,
    notes: "ultrices posuere cubilia Curae; Donec"
  },
  {
    personName: "Asher Puckett",
    contract: "НЕТ",
    dateBirth: "18-09-1963",
    telNum: "1-690-233-3770",
    code: "40",
    autoMonth: 9,
    remain: 4,
    days: "24-01-2020",
    photoId: 0,
    rent: "07-12-2019",
    deposite: 4218,
    notes: "nisi. Aenean eget metus. In"
  },
  {
    personName: "Miriam Levine",
    contract: "СОТРУДНИК",
    dateBirth: "09-08-1959",
    telNum: "1-783-837-2369",
    code: "41",
    autoMonth: 8,
    remain: 9,
    days: "06-05-2019",
    photoId: 0,
    rent: "20-10-2020",
    deposite: 769,
    notes: "vestibulum. Mauris magna. Duis dignissim"
  },
  {
    personName: "Arsenio Hendricks",
    contract: "СОТРУДНИК",
    dateBirth: "16-08-1970",
    telNum: "1-152-271-3796",
    code: "42",
    autoMonth: 2,
    remain: 8,
    days: "31-07-2020",
    photoId: 0,
    rent: "11-03-2020",
    deposite: -196,
    notes: "adipiscing non, luctus sit amet,"
  },
  {
    personName: "Leilani Wheeler",
    contract: "СОТРУДНИК",
    dateBirth: "27-01-1973",
    telNum: "1-739-409-1958",
    code: "43",
    autoMonth: 1,
    remain: 13,
    days: "13-01-2021",
    photoId: 0,
    rent: "19-08-2020",
    deposite: 2852,
    notes: "interdum. Sed auctor odio a"
  },
  {
    personName: "Christen Sloan",
    contract: "6 ТА",
    dateBirth: "20-06-1968",
    telNum: "1-507-644-9918",
    code: "44",
    autoMonth: 4,
    remain: "",
    days: "12-11-2019",
    photoId: 0,
    rent: "29-12-2020",
    deposite: 3521,
    notes: "nunc, ullamcorper eu, euismod ac,"
  },
  {
    personName: "Lev Hess",
    contract: "СОТРУДНИК",
    dateBirth: "12-10-1955",
    telNum: "1-262-360-4575",
    code: "45",
    autoMonth: 7,
    remain: "",
    days: "10-06-2020",
    photoId: 0,
    rent: "10-06-2019",
    deposite: 4461,
    notes: "malesuada vel, venenatis vel, faucibus"
  },
  {
    personName: "Jasper Mills",
    contract: "3 КФ",
    dateBirth: "10-12-1994",
    telNum: "1-634-953-8909",
    code: "46",
    autoMonth: 3,
    remain: "",
    days: "29-12-2019",
    photoId: 0,
    rent: "07-10-2019",
    deposite: 755,
    notes: "consectetuer adipiscing elit. Curabitur sed"
  },
  {
    personName: "Hyatt Johnston",
    contract: "ПОСЕЩЕНИЯ",
    dateBirth: "04-12-2004",
    telNum: "1-555-735-5064",
    code: "47",
    autoMonth: 8,
    remain: 10,
    days: "25-06-2019",
    photoId: 0,
    rent: "18-01-2021",
    deposite: 2133,
    notes: "arcu. Sed et libero. Proin"
  },
  {
    personName: "Gloria Dixon",
    contract: "ПОСЕЩЕНИЯ",
    dateBirth: "02-07-1952",
    telNum: "1-553-504-4377",
    code: "48",
    autoMonth: 5,
    remain: 18,
    days: "26-11-2020",
    photoId: 0,
    rent: "09-09-2019",
    deposite: -77,
    notes: "eget varius ultrices, mauris ipsum"
  },
  {
    personName: "Jonah Dunlap",
    contract: "ПОСЕЩЕНИЯ",
    dateBirth: "20-06-1953",
    telNum: "1-967-741-9273",
    code: "49",
    autoMonth: 5,
    remain: 18,
    days: "04-10-2020",
    photoId: 0,
    rent: "15-02-2020",
    deposite: 2028,
    notes: "diam. Proin dolor. Nulla semper"
  },
  {
    personName: "Wesley White",
    contract: "НЕТ",
    dateBirth: "23-04-1993",
    telNum: "1-522-349-9026",
    code: "50",
    autoMonth: 5,
    remain: 24,
    days: "30-12-2019",
    photoId: 0,
    rent: "23-03-2019",
    deposite: 3865,
    notes: "ut ipsum ac mi eleifend"
  }
];
