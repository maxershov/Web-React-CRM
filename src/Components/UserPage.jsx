/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable import/no-cycle */
import React, { useState } from 'react';
import { connect } from 'react-redux';
import { useParams, useHistory } from 'react-router-dom'
import { deletePerson } from '../App';
import FormData from './FormData';
import CalendarHideable from './CalendarHideable';
import FieldDeposite from './FieldDeposite';
import FieldsAction from './FieldsAction';
import TableHistory from './TableHistory';
import AreaNotes from './AreaNotes';


function getAllPersonNames(data) {
  return data.map(obj => { return obj.personName });
}

export const UserPage = (props) => {
  /** Show fields with User data and change it on submit */
  const [renderPhotoId, changeRenderPhotoId] = useState(false);
  const { codeLink } = useParams();
  const history = useHistory();

  const data = JSON.parse(props.personData);

  let person = data.find(profile => {
    return profile.code === codeLink
  })

  if (person === undefined) {
    // TODO kill this CRAP => prevent from crash if unvalid code in URL path
    [person,] = data;
    history.push('/main');
  }



  let renderFields = '';
  if (person.contract === 'ЛИД') { renderFields = <LeadParams person={person} /> }
  else if (person.contract === 'СОТРУДНИК') { renderFields = <EmployeeParams person={person} /> }
  else if (person.contract === 'НЕТ') { renderFields = <LostPersonParams person={person} /> }
  else { renderFields = <PersonParams person={person} /> }

  return (
    <div className="userPage">
      <div className="img-container"><img onClick={() => changeRenderPhotoId(!renderPhotoId)} alt="profilePhoto" src={require(`../images/${person.photoId}.jpg`)} /></div>
      <div className="userPage-container">
        {renderPhotoId ? (
          <>
            <FormData className="photoId" formLabel="Изменить код фото:" baseValue={person.photoId} inputType="photoId" type="PERSON" />
            <label>Удаление:</label>
            <button type="button" onClick={() => deletePerson(person.code)}>Удалить пользователя</button>
          </>
        ) : undefined}
        <FormData formLabel="Имя:" baseValue={person.personName} inputType="personName" type="PERSON" />
        <FormData formLabel="Номер телефона:" baseValue={person.telNum} inputType="telNum" type="PERSON" />
        {renderFields}
      </div>
      <div className="notesField">
        <label>Заметки:</label>
        <AreaNotes notesValue={person.notes} type="PERSON" />
      </div>
      <FieldsAction code={person.code} namesArr={getAllPersonNames(data)} />
      <TableHistory tableDataType="personData" code={person.code} />
    </div>
  );
}

export const PersonParams = (props) => {
  const { person } = props;
  return (
    <>
      <FormData formLabel="Дата рождения:" baseValue={person.dateBirth} inputType="dateBirth" type="PERSON" />
      <FormData formLabel="Абонемент:" baseValue={person.contract} inputType="contract" type="PERSON" />
      <FormData formLabel="Остаток тренировок:" baseValue={person.remain} inputType="remain" type="PERSON" />
      <FormData formLabel="Код карты:" baseValue={person.code} inputType="code" type="PERSON" />
      <FormData formLabel="Месяц парковки:" baseValue={person.autoMonth} inputType="autoMonth" type="PERSON" />
      <CalendarHideable сalendarName="Срок контракта:" dateType="days" date={person.days} />
      <CalendarHideable сalendarName="Срок аренды шкафа:" dateType="rent" date={person.rent} />
      <FieldDeposite depositeValue={person.deposite} />
    </>
  )
}


export const LeadParams = (props) => {
  const { person } = props;
  return (
    <>
      <FormData formLabel="Тип профиля:" baseValue={person.contract} inputType="contract" type="PERSON" />
      <CalendarHideable сalendarName="Дата первого обращения:" dateType="rent" date={person.rent} />
    </>
  )
}

export const LostPersonParams = (props) => {
  const { person } = props;
  return (
    <>
      <FormData formLabel="Тип профиля:" baseValue={person.contract} inputType="contract" type="PERSON" />
      <FormData formLabel="Дата рождения:" baseValue={person.dateBirth} inputType="dateBirth" type="PERSON" />
      <FormData formLabel="Код карты:" baseValue={person.code} inputType="code" type="PERSON" />
      <FieldDeposite depositeValue={person.deposite} />
      <CalendarHideable сalendarName="Дата окончания контракта:" dateType="days" date={person.days} />
    </>
  )
}


export const EmployeeParams = (props) => {
  const { person } = props;
  return (
    <>
      <FormData formLabel="Дата рождения:" baseValue={person.dateBirth} inputType="dateBirth" type="PERSON" />
      <FormData formLabel="Тип профиля:" baseValue={person.contract} inputType="contract" type="PERSON" />
      <FormData formLabel="Код карты:" baseValue={person.code} inputType="code" type="PERSON" />
      <FormData formLabel="Парковка оплачена до:" baseValue={person.autoMonth} inputType="autoMonth" type="PERSON" />
      <FieldDeposite depositeValue={person.deposite} />
    </>
  )
}

const mapStateToProps = state => {
  return {
    personData: state.personStore.data
  }
}

export default connect(mapStateToProps)(UserPage);