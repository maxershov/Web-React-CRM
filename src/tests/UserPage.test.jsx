import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { UserPage, PersonParams, LeadParams, LostPersonParams, EmployeeParams } from '../Components/UserPage'
import { person, lead, employee, lost } from './testData/testData'


// Set mock for router func => get url for navigation
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn(),
  }),
  useParams: () => ({
    codeLink: '111111',
  }),
}));

// Set default date to make all snapshots equal
const defaultDate = new Date(0);
jest.spyOn(global, 'Date').mockImplementation(() => defaultDate);


test('should render UserPage person correctly', () => {
  const wrapper = shallow(<UserPage personData={JSON.stringify(person)} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('should render PersonParams correctly', () => {
  const wrapper = shallow(<PersonParams person={JSON.stringify(person[0])} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});


test('should render LeadParams correctly', () => {
  const wrapper = shallow(<LeadParams person={JSON.stringify(lead[0])} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('should render LostPersonParams correctly', () => {
  const wrapper = shallow(<LostPersonParams person={JSON.stringify(lost[0])} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('should render EmployeeParams correctly', () => {
  const wrapper = shallow(<EmployeeParams person={JSON.stringify(employee[0])} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});