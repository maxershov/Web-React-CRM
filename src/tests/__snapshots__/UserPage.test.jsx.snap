// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EmployeeParams correctly 1`] = `
<Fragment>
  <FormData
    formLabel="Дата рождения:"
    inputType="dateBirth"
    type="PERSON"
  />
  <FormData
    formLabel="Тип профиля:"
    inputType="contract"
    type="PERSON"
  />
  <FormData
    formLabel="Код карты:"
    inputType="code"
    type="PERSON"
  />
  <FormData
    formLabel="Парковка оплачена до:"
    inputType="autoMonth"
    type="PERSON"
  />
  <FieldDeposite />
</Fragment>
`;

exports[`should render LeadParams correctly 1`] = `
<Fragment>
  <FormData
    formLabel="Тип профиля:"
    inputType="contract"
    type="PERSON"
  />
  <CalendarHideable
    dateType="rent"
    сalendarName="Дата первого обращения:"
  />
</Fragment>
`;

exports[`should render LostPersonParams correctly 1`] = `
<Fragment>
  <FormData
    formLabel="Тип профиля:"
    inputType="contract"
    type="PERSON"
  />
  <FormData
    formLabel="Дата рождения:"
    inputType="dateBirth"
    type="PERSON"
  />
  <FormData
    formLabel="Код карты:"
    inputType="code"
    type="PERSON"
  />
  <FieldDeposite />
  <CalendarHideable
    dateType="days"
    сalendarName="Дата окончания контракта:"
  />
</Fragment>
`;

exports[`should render PersonParams correctly 1`] = `
<Fragment>
  <FormData
    formLabel="Дата рождения:"
    inputType="dateBirth"
    type="PERSON"
  />
  <FormData
    formLabel="Абонемент:"
    inputType="contract"
    type="PERSON"
  />
  <FormData
    formLabel="Остаток тренировок:"
    inputType="remain"
    type="PERSON"
  />
  <FormData
    formLabel="Код карты:"
    inputType="code"
    type="PERSON"
  />
  <FormData
    formLabel="Месяц парковки:"
    inputType="autoMonth"
    type="PERSON"
  />
  <CalendarHideable
    dateType="days"
    сalendarName="Срок контракта:"
  />
  <CalendarHideable
    dateType="rent"
    сalendarName="Срок аренды шкафа:"
  />
  <FieldDeposite />
</Fragment>
`;

exports[`should render UserPage person correctly 1`] = `
<div
  className="userPage"
>
  <div
    className="img-container"
  >
    <img
      alt="profilePhoto"
      onClick={[Function]}
      src={Object {}}
    />
  </div>
  <div
    className="userPage-container"
  >
    <FormData
      baseValue="Иванов Иван Иванович"
      formLabel="Имя:"
      inputType="personName"
      type="PERSON"
    />
    <FormData
      baseValue=""
      formLabel="Номер телефона:"
      inputType="telNum"
      type="PERSON"
    />
    <PersonParams
      person={
        Object {
          "autoMonth": "02",
          "code": "111111",
          "contract": "12 мес",
          "dateBirth": "29-02-1980",
          "days": "10-10-2021",
          "deposite": 2000,
          "notes": "Заметка",
          "personName": "Иванов Иван Иванович",
          "photoId": 0,
          "remain": "",
          "rent": "10-02-2021",
          "telNum": "",
        }
      }
    />
  </div>
  <div
    className="notesField"
  >
    <label>
      Заметки:
    </label>
    <AreaNotes
      notesValue="Заметка"
      type="PERSON"
    />
  </div>
  <MyActionFields
    code="111111"
    namesArr={
      Array [
        "Иванов Иван Иванович",
      ]
    }
  />
  <Connect(TableHistory)
    code="111111"
    tableDataType="personData"
  />
</div>
`;
